python/tvm/relax/frontend/onnx_frontend.py

```diff
@@ -531,6 +537,100 @@ def _parse_attr(self, attr_proto):
                 raise ValueError("Cannot parse attribute: \n{}\n.".format(a))
         return attrs
 
+    def _relay_input_adapter(self, inputs):
+        """Creates equivalent input Relay vars from the input Relax vars"""
+        relay_vars = []
+        for relax_var in inputs:
+            shape_values = []
+            for shape_value in relax_var.struct_info.shape.values:
+                shape_values.append(shape_value)
+            if isinstance(relax_var, relax.Constant):
+                relay_vars.append(relay.const(relax_var.data, dtype=relax_var.checked_type.dtype))
+            else:
+                relay_vars.append(relay.var(relax_var.name_hint, shape=shape_values, dtype=relax_var.checked_type.dtype))
+        return relay_vars
+
+
+    def _relay_output_adapter(self, relax_inputs, relay_inputs, relay_output):
+        """Given the output of a relay op from the Onnx relay frontend,
+        calls into the relay to relax translator to obtain the equivalent Relax.
+        Then unpacks the IRModule obtained and adds the TIR funcs and the
+        associated call_tirs to the block builder in use.
+        Parameters
+        ----------
+        relax_inputs : list(relax.Var, relay.Constant)
+                The list of relax vars that are inputs to the relax op.
+        relay_inputs : list(relay.Var, relay.Constant)
+                The list of relay vars that are inputs to the relay op. This is
+                obtianed from the _relay_input_adapter function.
+        relay_output : relay.Expr
+                The output of the relay op from the Onnx relay frontend.
+        Returns
+        -------
+        output : relax.Expr
+                The output of the equivalent relax op.
+        """
+        if isinstance(relay_output, TupleWrapper):
+            relay_output = relay_output.tuple_value
+
+        # Create a Relay function with the body returned by the Relay op.
+        relay_var_inputs = [input for input in relay_inputs if isinstance(input, relay.Var)]
+        function = relay.Function(relay_var_inputs, relay_output)
+        # TODO Should we support passing a target into the frontend?
+        target = Target("llvm")
```

Comment:
potentially yes we should support passing other targets (e.g. cuda). it's a bit uncomfortable making these sort of decisions in the frontend, but i dont think we have much choice while we don't have Relax op lowering in place.